lib/src/rules/required.ts

```diff
@@ -17,7 +17,7 @@ export const required: FieldValidationFunction = (value, vm, customParams: Requi
 function isValidField(value, trim: boolean): boolean {
   return typeof value === 'string' ?
     isStringValid(value, trim) :
-    value === true;
+    value === true || typeof value === "number";
```

Comment:
Extract `value === true || typeof value === "number"` in a function like `isNonStringValid` to check non string values. That is, is different than `undefined` or `null`

Reply:
Evaluate if value is different than null or undefined collides with the "When validating a non string value should return false if value is false" test. What do you consider for a boolean with false value?